### ISSUE TITLE
Rule NoSemicolons does not work

### ISSUE THREAD

**Issue by rumatoest** [arturbosch/detekt#1299]
Using latest RC9.2

```kotlin
try {
    return stuff(); // Actually there is semicolon here
} catch (ex: Throwable) {}
```
In configuration I have
```
  NoSemicolons:
    active: true
    autoCorrect: true
```

But it does not report about errors

**Comment by schalkms** [comment]
Thanks for reporting this.
This rule originates from [ktlint](https://github.com/shyiko/ktlint). Detekt simply provides a wrapper around this ruleset. I ask you to report this issue in the ktlint repo since there's not much detekt's team can do about this.

**Comment by vanniktech** [comment]
And which is why this wrapping functionality from Detekt shouldn't even exist in the first place.

**Comment by rumatoest** [comment]
>there's not much detekt's team can do about this.
Actually you can. 
NoSemicolon rule in ktlint works perfectly, so I think this is Detekt issue because this behavior  has to be configurable according to Detekt docs https://arturbosch.github.io/detekt/formatting.html

**Comment by schalkms** [comment]
> NoSemicolon rule in ktlint works perfectly

True. That information was important. I can verify this. It seems like the comment after the semicolon is causing problems. If I remove the comment, the semicolon gets removed.

**Comment by rumatoest** [comment]
> If I remove the comment, the semicolon gets removed.

Actually in my project there is not comments and semicolon does not removed nor detected as error

**Comment by arturbosch** [comment]
Just run `detekt` with `detekt-formatting` on detekt with two introduced semicolons and they got removed.
Maybe its confusing that the default detekt config was the formatting rule set in it while it still needs the formatting jar ...

@vanniktech maybe it is your opinion. But actually it makes sense to wrap ktlint rules:
1. do not recreate the expensive `KotlinCoreEnvironment` for KtLint
2. have ktlint warnings integrated into sonar-kotlin or other tools like codacy (and private tools) without writing the integration itself.

1. is actually huge. most detekt users just use the gradle plugin. Each task per module needs the ~500-800ms creation overhead of the `KotlinCoreEnvironment`. And thats just for 20k lines.

**Comment by Mauin** [comment]
Opened #1304 for the overall ktlint discussion so that this issue can focus on the problem at hand.

**Comment by schalkms** [comment]
I'm closing this, since the discussion about this issue has been moved.
